entrypoint.sh

```diff
@@ -19,46 +31,45 @@ process_input(){
                 scan_file "$path"
             fi
         done
-        if [ -z "$execution_mode" ]; then exit $status_code; fi
+        [ -z "$execution_mode" ] && exit $status_code
     else
         scan_all "$my_dir"
-        if [ -z "$execution_mode" ]; then exit $status_code; fi
+        [ -z "$execution_mode" ] && exit $status_code
     fi
 }
 
 scan_file(){
     local file_path=$1
     local file=$(basename -- "$file_path")
-    local extension="${file##*.}"
-
-    if [ "$extension" == "sh" ]; then
+    local first_line=$(head -n 1 "$file_path")
+    if [[ -x "$file_path" ]] && [[ "$first_line" == "#!"* ]]; then
         echo
         echo "###############################################"
         echo "         Scanning $file"
         echo "###############################################"
         shellcheck "$file_path"
         local exit_code=$?
         if [ $exit_code -eq 0 ] ; then
-            echo "Successfully scanned ${file_path} 🙌"
+            printf "%b" "Successfully scanned ${file_path} 🙌\n"
         else
             status_code=$exit_code
             printf "\e[31m \n ERROR: ShellCheck detected issues in %s.\e[0m \n" "${file_path} 🐛"
         fi
     else
-        printf "\e[33m ⚠️  Warning: invalid file extension. Make sure the input file '%s' is a valid shell script.\e[0m\n" "$file"
+        printf "\e[33m ⚠️  Warning: '%s' is not a valid shell script. Make sure file is executable & shebang is on the first line.\e[0m\n" "$file_path"
     fi
 }
 
 scan_all(){
-    echo "Scanning all the scripts with format name.sh at $1 🔎"
+    echo "Scanning all the shell scripts at $1 🔎"
     while IFS= read -r -d $'\0' script 
     do
         scan_file "$script"
-    done < <(find "$1" -name '*.sh' -print0)
+    done < <(find "$1" -type f -perm ${perm_arg} ! -path "$1/.git/*" -print0 )
```

Comment:
For scan all, we are finding all the files in a given folder ($1). Why are we specifying the path?

Reply:
Now that we lint scripts with no extension, the default case will lint all the scripts in the .git directory as well. And we don't want that, that's why I'm excluding the `.git` directory.